src/app/modules/group/http-services/get-requests.service.ts

```diff
@@ -3,10 +3,60 @@ import { HttpClient, HttpParams } from '@angular/common/http';
 import { Observable } from 'rxjs';
 import { appConfig } from 'src/app/shared/helpers/config';
 import { map } from 'rxjs/operators';
+import { pipe } from 'fp-ts/function';
+import * as D from 'io-ts/Decoder';
+import { decodeSnakeCase } from '../../../shared/operators/decode';
+import { dateDecoder } from '../../../shared/helpers/decoders';
+
+const userDecoder = pipe(
+  D.struct({
+    groupId: D.string,
+    login: D.string,
+  }),
+  D.intersect(
+    D.partial({
+      firstName: D.nullable(D.string),
+      lastName: D.nullable(D.string),
+      grade: D.nullable(D.number),
+    }),
+  ),
+);
+
+const groupPendingRequestDecoder = pipe(
+  D.struct({
+    at: D.nullable(dateDecoder),
+    group: D.struct({
+      id: D.string,
+      name: D.string,
+    }),
+    type: D.literal('join_request', 'leave_request'),
+    user: userDecoder,
+  }),
+);
+
+const groupInvitationDecoder = pipe(
+  D.struct({
+    action: D.literal('invitation_created', 'join_request_created', 'join_request_refused'),
+    at: dateDecoder,
+    group: D.struct({
+      id: D.string,
+      name: D.string,
+      description: D.nullable(D.string),
+      type: D.literal('Class', 'Team', 'Club', 'Friends', 'Other', 'Session'),
+    }),
+    groupId: D.string,
+    invitingUser: D.nullable(D.struct({
+      id: D.string,
+      firstName: D.string,
+      lastName: D.string,
+      login: D.string,
+    })),
+  })
+);
 
 export interface PendingRequest {
```

Comment:
I think some of the decoders can be reused as types in the interfaces below, there is a lot of type-related code duplicate here

Reply:
Actually I tried, but realised it requires a lot of changes including group-manager-list.component.spec.ts 

Maybe it could be done in separated issue?

Reply:
I think it's fine as is, I think it won't be worth the effort. Let's leave it like this.